File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/dto/User/UserDto.java
package com.ntnu.gidd.dto.User;

import com.ntnu.gidd.util.TrainingLevelEnum;
import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDto extends ContextAwareUserDto {
	@NotNull
	private UUID id;
	@NotNull
	@NotEmpty
	private String firstName;
	@NotNull
	@NotEmpty
	private String surname;
	@NotNull
	@NotEmpty
	private String email;
	private LocalDate birthDate;
	private TrainingLevelEnum level;
	private String image;
	private Integer followingCount;
	private Integer followerCount;

}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/user/dto/UserDto.kt
package ntnu.idatt2105.user.dto

import ntnu.idatt2105.user.model.User
import java.time.LocalDate
import java.util.*

data class UserDto(
    val id: UUID = UUID.randomUUID(),
    val firstName: String = "",
    val surname: String = "",
    val email: String = "",
    val phoneNumber: String = "",
    val image: String = "",
    val expirationDate: LocalDate = LocalDate.EPOCH
)

fun User.toUserDto() = UserDto(id = this.id, firstName = this.firstName, surname = this.surname, email = this.email,
        phoneNumber = this.phoneNumber, image = this.image, expirationDate = this.expirationDate)
